lib/public_api.ts

```diff
@@ -9,32 +9,184 @@
  * from here.
  */
 
+import { awaitConnectionFromIframe } from "./messaging";
+import {
+  FledgeRequest,
+  isRunAdAuctionResponse,
+  RequestTag,
+} from "./shared/protocol";
+
+export { VERSION } from "./shared/version";
+
+// The FLEDGE spec uses snake_case property names.
+/* eslint-disable camelcase */
+
 /**
- * Creates a new registration in this browser for a specified interest group and
- * stores it client-side.
+ * An ad creative that can participate in an auction and later be rendered onto
+ * the page if it wins.
  *
- * @see https://github.com/WICG/turtledove/blob/main/FLEDGE.md#11-joining-interest-groups
+ * The properties of this type aren't actually specified in the FLEDGE spec at
+ * present; they are our best guess as to how this will work, and may be
+ * replaced later with a different API.
  */
-export function joinAdInterestGroup(): void {
-  throw new Error("Not yet implemented");
+export interface Ad {
+  /**
+   * The URL where the actual creative is hosted. This will be used as the `src`
+   * of an iframe that will appear on the page if this ad wins.
+   */
+  rendering_url: string;
+  /** Additional metadata about this ad that can be read by the auction. */
+  metadata: {
+    /**
+     * The amount in USD that the buyer is willing to pay per thousand rendered
+     * impressions of this ad.
+     *
+     * This is used by our temporary hardcoded auction logic and will not exist
+     * in browser-native implementations of FLEDGE (in which auction logic, and
+     * the structure and semantics of ad metadata, will be user-defined).
```

Comment:
This comment is exactly how we should do it: when we have to make a simplification that's fine, we just need to say that we're departing from the spec